ssl/t1_lib.c

```diff
@@ -517,7 +521,7 @@ static int add_provider_sigalgs(const OSSL_PARAM params[], void *data)
 
     p = OSSL_PARAM_locate_const(params, OSSL_CAPABILITY_TLS_SIGALG_MIN_TLS);
     if (p == NULL || !OSSL_PARAM_get_int(p, &sinf->mintls) ||
-		    sinf->mintls < TLS1_3_VERSION) {
+		    sinf->mintls != TLS1_3_VERSION) {
```

Comment:
Would it be better to ignore sigalgs claiming support for TLS versions that we can't handle? Maybe this provider is written for some future version of OpenSSL that supports pluggable sigalgs for TLS1.2 (or even TLS1.4) - but its backwards compatible with OpenSSL 3.2.

For DTLS I guess we always ignore these sigalgs? Which is probably the correct behaviour.

Shouldn't we also validate that maxtls > mintls (if maxtls != 0 && maxtls != -1)? If one of maxtls or mintls is set to -1, both should be I think.

Reply:
Added in https://github.com/openssl/openssl/pull/19312/commits/8be6985a9743a239684308b6258691d3489e52bf

Reply:
I think you missed the first part of my comment here, i.e. "Would it be better to ignore sigalgs claiming support for TLS versions that we can't handle?"

Reply:
I thought I didn't miss that comment (by now _requiring_ TLS1_3_VERSION -- which is guaranteed to be handled when this define is set, right?). But re-reading your comment again, I now see your point (with different OpenSSL variants supporting different TLS versions). So assuming there never will be an OpenSSL with "missing" TLS versions (say TLS1.1 and TLS1.3 _is_ supported but TLS1.2 is _not_), would you be OK with a test for mintls>=TLS1_VERSION and maxtls <=TLS_MAX_VERSION?

Reply:
See corresponding code update in https://github.com/openssl/openssl/pull/19312/commits/0b151b0e8bb86c140eb8b5ee932b60f07f986b31

Reply:
What changes are still expected to consider this resolved?

Reply:
I don't think is quite correct yet. But the code has moved on a bit since this comment was made so I'll make new comments against the current state of the code and resolve this thread.